pkg/controller/replication/replication_controller.go

```diff
@@ -579,13 +579,26 @@ func (rm *ReplicationManager) syncReplicationController(key string) error {
 	if err != nil {
 		return err
 	}
-	// Ignore inactive pods.
-	var filteredPods []*v1.Pod
+	// Filter out active and rejected pods.
+	var filteredActivePods, filteredRejectedPods []*v1.Pod
 	for _, pod := range allPods {
+		if controller.IsPodRejected(pod) {
+			filteredRejectedPods = append(filteredRejectedPods, pod)
+		}
 		if controller.IsPodActive(pod) {
-			filteredPods = append(filteredPods, pod)
+			filteredActivePods = append(filteredActivePods, pod)
 		}
 	}
+
+	// If there are pods rejected by kubelet admit, delete them and return an error to ensure rate limit.
+	if len(filteredRejectedPods) > 0 {
+		for _, pod := range filteredRejectedPods {
+			// Ignore deletion error here.
+			rm.podControl.DeletePod(pod.Namespace, pod.Name, rc)
+		}
+		return fmt.Errorf("rejected pods exist, delete them and exit sync!")
```

Comment:
I don't think this will help much because this will ratelimit the next sync (5ms) but the next sync will be successful thus you will end up in a 5ms-now-5ms-now-... hotloop.